packages/aws-cdk-lib/aws-efs/lib/efs-file-system.ts

```diff
@@ -409,7 +491,26 @@ export class FileSystem extends FileSystemBase {
       throughputMode: props.throughputMode,
       provisionedThroughputInMibps: props.provisionedThroughputPerSecond?.toMebibytes(),
       backupPolicy: props.enableAutomaticBackups ? { status: 'ENABLED' } : undefined,
-      fileSystemPolicy: Lazy.any({ produce: () => this._fileSystemPolicy }),
+      fileSystemPolicy: Lazy.any({
+        produce: () => {
+          const allowAnonymousAccess = props.allowAnonymousAccess ?? !this._grantedClient;
```

Comment:
I think this should be the default behavior. Would you mind/be able to create a
new feature flag and make this the new default?

Reply:
Does that mean that the behavior of deny anonymous clients should be made the default even when not using `grantClient`?

Reply:
yes, but only with the feature flag.